src/mds/MDBalancer.cc

```diff
@@ -150,6 +150,15 @@ void MDBalancer::handle_export_pins(void)
 
   set<CDir *> authsubs;
   mds->mdcache->get_auth_subtrees(authsubs);
+
+  if (authsubs.size() > AUTH_TREES_THRESHOLD) {
+    dout(15) << "number of auth tree is beyond threshold" << dendl;
+    dout(15) << "number of auth trees = " <<  authsubs.size() << dendl;
+    dout(15) << "AUTH_TREES_THRESHOLD = " << AUTH_TREES_THRESHOLD << dendl;
```

Comment:
Collapse this into a single short message. These high debug (15) messages do not need to be easily readable. As long as you can find it in the source, it's easy enough to figure it out.